ext/telemetry/lib.rs

```diff
@@ -866,9 +969,596 @@ fn op_otel_span_flush(state: &mut OpState) {
     return;
   };
 
-  let Some((span_processor, _)) = OTEL_PROCESSORS.get() else {
+  let Some(Processors { spans, .. }) = OTEL_PROCESSORS.get() else {
+    return;
+  };
+
+  spans.on_end(temporary_span.0);
+}
+
+pub struct TemporaryMetricsExport {
```

Comment:
Could you add some docstrings here about the purpose of these structs and enums?